test-data/unit/check-varargs.test

```diff
@@ -1066,3 +1117,166 @@ class C:
 [builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsAssignmentWhenSourceIsUntyped]
+from typing_extensions import Unpack, TypedDict
+
+class Movie(TypedDict):
+    name: str
+    year: int
+
+def src(**kwargs): ...
+def dest(**kwargs: Unpack[Movie]): ...
+
+dest = src  # OK
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsAssignmentWhenSourceHasTraditionallyTypedKwargs]
+from typing_extensions import Unpack, TypedDict
+
+class Vehicle:
+    ...
+
+class Car(Vehicle):
+    ...
+
+class Motorcycle(Vehicle):
+    ...
+
+def src(**kwargs: Vehicle): ...
+
+class Vehicles(TypedDict):
+    car: Car
+    moto: Motorcycle
+
+def dest(**kwargs: Unpack[Vehicles]): ...
+
+dest = src  # OK
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsAssignmentWhenDestinationHasTraditionallyTypedKwargs]
+from typing_extensions import Unpack, TypedDict
+
+class Vehicle:
+    ...
+
+class Car(Vehicle):
+    ...
+
+class Motorcycle(Vehicle):
+    ...
+
+def dest(**kwargs: Vehicle): ...
+
+class Vehicles(TypedDict):
+    car: Car
+    moto: Motorcycle
+
+def src(**kwargs: Unpack[Vehicles]): ...
+
+dest = src  # E: Incompatible types in assignment (expression has type "Callable[[KwArg(Vehicles)], Any]", variable has type "Callable[[KwArg(Vehicle)], Any]")
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsAssignmentWhenOnlySourceContainsKwargs]
+from typing import NotRequired
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+class Example(TypedDict):
+    animal: Animal
+    string: str
+    number: NotRequired[int]
+
+def src(**kwargs: Unpack[Example]): ...
+def dest(*, animal: Dog, string: str, number: int = ...): ...
+
+dest = src  # OK
+
+def dest_with_positional_args(animal: Dog, string: str, number: int = ...): ...
+
+dest_with_positional_args = src  # E: Incompatible types in assignment (expression has type "Callable[[KwArg(Example)], Any]", variable has type "Callable[[Dog, str, int], Any]")
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testUnpackKwargsAssignmentWhenOnlyDestinationContainsKwargs]
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+def dest(**kwargs: Unpack[Animal]): ...
+def src(name: str): ...
+
+dog: Dog = {"name": "Daisy", "breed": "Labrador"}
+animal: Animal = dog
+
+dest = src  # E: Incompatible function signatures - variable contains **kwargs and the expression does not. See PEP 692 for more details.
+
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testUnpackKwargsAssignmentWhenBothSidesContainKwargsTypedWithUnpack]
+from typing_extensions import Unpack, TypedDict
+
+class Animal(TypedDict):
+    name: str
+
+class Dog(Animal):
+    breed: str
+
+def accept_animal(**kwargs: Unpack[Animal]): ...
+def accept_dog(**kwargs: Unpack[Dog]): ...
+
+accept_dog = accept_animal  # OK
```

Comment:
You are right, when I was writing this I was thinking more in terms of "will this cause a runtime error or not?" - but in terms of "is this a subtype of that?" it's not correct. Something I'll need to revisit.

Reply:
I agree that it is true, but only if we look at it from the normalized signature point of view. My thinking is that what matters is what is happening later in the function itself. If the function contains `**kwargs`, then it means inside the function we can use a variable `kwargs` - wouldn't in that case thinking of `**kwargs` as just another parameter that should just adhere to the contravariance of parameters rule make sense? 

It becomes very cumbersome in case of "normal" function signatures (without `**kwargs`) interacting with the functions that use `**kwargs`...

Reply:
There is nothing cumbersome in reducing the PEP to just one paragraph that would explain that `Unpack[SomeTD]` is a syntactic sugar for (and is considered equivalent to) the expanded signature. This has a number of benefits:
* This will not add any new unsafety that is not already present for existing uses of TypedDicts in `**` contexts. (And type checkers may handle this unsafety in a uniform way, say in mypy we *may* use existing `--extra-checks` flag to prohibit some techincally unsafe calls as I mentioned before.)
* This is actually easy to remember and to reason about.
* This will allow people who want subtyping between callables to easily achieve this using `total=False`, which follows from existing rules for expanded callables.

Btw a comment on the last point, subtyping relations are quite counter-intuitive when you think about them in terms of packed TypedDicts. Consider this example:
```python
class A(TypedDict, total=False):
    x: int
class B(TypedDict, total=False):
    x: int
    y: int

def takes_a(**kwargs: Unpack[A]) -> None: ...
def takes_b(**kwargs: Unpack[B]) -> None: ...

if bool():
    takes_a = takes_b  # mypy now says it is OK
else:
    takes_b = takes_a  # mypy now gives an error for this
```
If you would think in terms of subtyping between TypedDicts, you might conclude the _opposite relations_, because `B` is subtype of `A`, and functions should be contravariant in argument types, right? But obviously `takes_a` is _not_ a subtype of `takes_b` because `takes_b(x=42, y=0)` is a valid call, while `takes_a(x=42, y=0)` is not a valid one. (You can say this is because mappings are contravariant in keys, but covariant in values etc., but yet again, simple logic of always expanding makes all this unnecessary).

Finally, the simple logic of always expanding is more future-proof. For example, if we would decide to add default types for TypedDicts, it would naturally translate to trailing homogeneous `**kwargs` in expanded form. While if you treat packed and expanded forms differently (as PEP currently does), it would add many new corner cases.

Reply:
Thanks for a detailed response! There's a lot to unpack here (pun intended), but I think what you're saying makes sense.